core/dtn7/src/routing/erouting/ws_client.rs

```diff
@@ -0,0 +1,112 @@
+use super::Packet;
+use futures::channel::mpsc::{unbounded, UnboundedReceiver, UnboundedSender};
+use futures_util::{future, pin_mut, SinkExt, StreamExt};
+use log::info;
+use serde_json::Result;
+use std::str::FromStr;
+use tokio_tungstenite::connect_async;
+use tokio_tungstenite::tungstenite::Message;
+
+pub enum Command {
+    /// Requests a send of the given packet.
+    SendPacket(Packet),
+    /// Requests the closure of the client.
+    Close,
+}
+
+/// Represents the client session of a external router.
+pub struct Client {
+    ip: String,
+    port: i16,
+    cmd_receiver: UnboundedReceiver<Command>,
+    cmd_sender: UnboundedSender<Command>,
+    packet_out: UnboundedSender<Packet>,
+}
+
+/// Creates a new extern router client.
+///
+/// # Arguments
+///
+/// * `addr` - Address to connect to in format ip:port without any websocket url parts.
+/// * `packet_our` - Channel to which received packets will be passed
+///
+pub fn new(addr: &str, packet_out: UnboundedSender<Packet>) -> std::io::Result<Client> {
+    let parts: Vec<&str> = addr.split(':').collect();
+
+    if parts.len() != 2 {
+        return Err(std::io::Error::new(
```

Comment:
complex error creation, consider using thiserror with own type or just bail

Reply:
Saw it used in ``dtnd.rs`` and ``dtnconfig.rs`` so I used the same error. Should I still change it?